src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -235,15 +241,30 @@ private void AnalyzeDirectory(DirectoryInfo directoryPath, ReportFormat reportFo
                         {
                             logger.LogError("\t{failedFile}", failedFile);
                         }
+                        return 6;
                     }
+                    return exitCode;
                 }
             }
             catch (Exception exp)
             {
                 logger.LogError(GetExceptionMessage(exp));
+                return 1;
             }
         }
 
+        private int CalculateExitCode(int exitCode, int res)
+        {
+            if (exitCode == 6 || res == 6)
+                return 6;
+            else if ((exitCode == 5 && res >= 1 && res <= 4) || (res == 5 && exitCode >= 1 && exitCode <= 4))
+                return 6;
+            else if (res == 5)
```

Comment:
For the moment, there is no return 5 issue in the code in the CLI. Where do you think we should return 5?